stetho-repl-js/src/main/java/com/facebook/stetho/inspector/console/js/JavaScriptRuntimeConsole.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.console.js;
+
+import com.facebook.stetho.inspector.console.CLog;
+import com.facebook.stetho.inspector.protocol.module.Console.MessageLevel;
+import com.facebook.stetho.inspector.protocol.module.Console.MessageSource;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.Function;
+import org.mozilla.javascript.ScriptRuntime;
+import org.mozilla.javascript.Scriptable;
+import org.mozilla.javascript.ScriptableObject;
+import org.mozilla.javascript.annotations.JSFunction;
+
+public class JavaScriptRuntimeConsole extends ScriptableObject {
+
+  /**
+   * Serial version UID.
+   */
+  private static final long serialVersionUID = 1L;
+
+  private JavaScriptRuntimeConsole thePrototypeInstance = null;
+
+  /**
+   * The zero-parameter constructor.
+   *
+   * When Context.defineClass is called with this class, it will construct
+   * File.prototype using this constructor.
+   */
+  public JavaScriptRuntimeConsole() {
+    if (thePrototypeInstance == null) {
+      thePrototypeInstance = this;
+    }
+  }
+
+  public JavaScriptRuntimeConsole(ScriptableObject scope) {
+    setParentScope(scope);
+    Object ctor = ScriptRuntime.getTopLevelProp(scope, "Console");
```

Comment:
Ok, I found where I took this from [Rhino's Environment object](https://github.com/mozilla/rhino/blob/master/toolsrc/org/mozilla/javascript/tools/shell/Environment.java#L53). They setup the `environment` object in the same way (see [Rhino's Global's environment](https://github.com/mozilla/rhino/blob/master/toolsrc/org/mozilla/javascript/tools/shell/Global.java#L114).

The suspect that the upper case name is the prototype.